MonashNovaRover/nova
# Add battery component in the top bar

## Description

1. Added battery component in the top bar.

## Changelogs

1. Added battery component in top bar
2. Setup ros topic, message type in ros directory.
3. Setup topic store in redux directory.



## Screenshots

<img width="574" alt="Screenshot 2024-10-06 at 11 14 19 PM" src="https://github.com/user-attachments/assets/2dea6fcf-463d-4523-b753-0858edcb9f47">
<img width="574" alt="Screenshot 2024-10-06 at 11 14 36 PM" src="https://github.com/user-attachments/assets/77a3ad42-09e4-42c8-a234-aafe1cbd7899">
<img width="574" alt="Screenshot 2024-10-06 at 11 14 53 PM" src="https://github.com/user-attachments/assets/a289cbf7-dc6a-47b5-9ed7-6ad2d4720aa4">

## Steps to Test

1. Start GUI and RosBridge.
2. Run this command in a separate terminal:
`ros2 topic pub --once /battery_state sensor_msgs/msg/BatteryState "{
    header: {
      stamp: {
        sec: 0,
        nanosec: 0
      },
      frame_id: ""
    },
    voltage: 30.0,
    current: 0.0,
    charge: 0.0,
    capacity: 0.0,
    design_capacity: 0.0,
    percentage: 0.0,
    power_supply_status: 0,
    power_supply_health: 0,
    power_supply_technology: 0,
    present: false,
    cell_voltage: [],
    location: "",
    serial_number: ""
  }
  "`
3. Change the voltage value to see different battery levels displayed on the GUI.
